ush/get_obs.py

```diff
@@ -0,0 +1,1004 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import shutil
+import argparse
+import logging
+from pathlib import Path
+import datetime as dt
+from textwrap import dedent
+from pprint import pprint
+from math import ceil, floor
+import subprocess
+from python_utils import (
+    load_yaml_config,
+)
+
+
+def get_obs_arcv_hr(obtype, arcv_intvl_hrs, hod):
+    """
+    This file defines a function that, for the given observation type, obs
+    archive interval, and hour of day, returns the hour (counting from hour
+    zero of the day) corresponding to the archive file in which the obs file
+    for the given hour of day is included.
+
+    Note that for cumulative fields (like CCPA and NOHRSC, as opposed to
+    instantaneous ones like MRMS and NDAS), the archive files corresponding
+    to hour 0 of the day represent accumulations over the previous day.  Thus,
+    here, we never return an achive hour of 0 for cumulative fields.  Instead,
+    if the specified hour-of-day is 0, we consider that to represent the 0th
+    hour of the NEXT day (i.e. the 24th hour of the current day) and set the
+    archive hour to 24.
+
+    Args:
+        obtype:
+        The observation type.  A string.
+
+        arcv_intvl_hrs:
+        Time interval (in hours) between archive files.  An integer.  For example,
+        if the obs files are bundled into 6-hourly archives, then this will be
+        set to 6.  This must be between 1 and 24 and must divide evenly into 24.
+
+        hod:
+        The hour of the day.  An integer.  This must be between 0 and 23.  For
+        cumulative fields (CCPA and NOHRSC), hour 0 is treated as that of the
+        next day, i.e. as the 24th hour of the current day.
+
+    Returns:
+        arcv_hr:
+        The hour since the start of day corresponding to the archive file containing
+        the obs file for the given hour of day.  An integer.
+    """
+
+    valid_obtypes = ['CCPA', 'NOHRSC', 'MRMS', 'NDAS']
+    obtype_upper = obtype.upper()
+    if obtype_upper not in valid_obtypes:
+        msg = dedent(f"""
+            The specified observation type (after converting to upper case) is not
+            supported:
+                obtype_upper = {obtype_upper}
+            Valid observation types are:
+                {valid_obtypes}
+        """)
+        logging.error(msg)
+        raise Exception(msg)
```

Comment:
Always better to use a more specific exception where possible
```suggestion
        raise ValueError(msg)
```

Reply:
Fixed everywhere pertinent in the script.